warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -683,11 +715,40 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
                                         registerEventMetadata(eventMetadataRegistry, enrichmentFieldName, edgeValue, jexlPreconditions);
                                     }
                                 }
+                                
+                            }
+                        }
+                    }
+                    
+                } else {
+                    Set<String> sourceSubGroups = matchingGroups.get(sourceGroup) != null ? matchingGroups.get(sourceGroup) : mSource.keySet();
+                    Set<String> sinkSubGroups = matchingGroups.get(sinkGroup) != null ? matchingGroups.get(sinkGroup) : mSink.keySet();
+                    
+                    for (String sourceSubGroup : sourceSubGroups) {
```

Comment:
This is also the same as above, just using a different set for the sourceSubGroups and sinkSubGroups.  Refactoring is definitely in order here.